src/execution_plan/ops/op_aggregate.c

```diff
@@ -198,27 +198,24 @@ static Record _handoff
 		return NULL;
 	}
 
-	Record   r    = OpBase_CreateRecord((OpBase*)op);
-	Group   *g    = (Group*)HashTableGetVal(entry);
-	SIValue *keys = g->keys;
-
-	// add all projected keys to the Record
-	for(uint i = 0; i < op->key_count; i++) {
-		int rec_idx = op->record_offsets[i];
-		// non-aggregated expression
-		SIValue key = SI_ShareValue(keys[i]);
-		Record_Add(r, rec_idx, key);
-	}
+	Group *g = (Group*)HashTableGetVal(entry);
+	Record r = g->r;
+	g->r = NULL;
 
 	// compute the final value of all aggregate expressions and add to Record
 	for(uint i = 0; i < op->aggregate_count; i++) {
 		int rec_idx = op->record_offsets[i + op->key_count];
 		AR_ExpNode *exp = g->agg[i];
 
 		SIValue agg = AR_EXP_FinalizeAggregations(exp, r);
```

Comment:
finalize aggregation one time value returning delete persist

Reply:
Proved to be more difficult then expected, although for the most part e.g. `count`, `max`, `min` this change is meaningless, the only place where this can show benefit is with the `collect` function, which causes a redundant clone of the array.